src/web/components/App.tsx

```diff
@@ -330,55 +330,27 @@ export const App = ({ CAPI, NAV }: Props) => {
-                        commentCount={commentCount}
-                        commentPage={commentPage}
-                        commentPageSize={commentPageSize}
-                        commentOrderBy={commentOrderBy}
-                        isClosedForComments={isClosedForComments}
-                        discussionD2Uid={CAPI.config.discussionD2Uid}
-                        discussionApiClientHeader={
-                            CAPI.config.discussionApiClientHeader
-                        }
-                        enableDiscussionSwitch={
-                            CAPI.config.enableDiscussionSwitch
-                        }
-                        expanded={true}
-                        commentToScrollTo={hashCommentId}
-                        onPermalinkClick={handlePermalink}
-                    />
-                ) : (
-                    <Lazy margin={300}>
-                        <CommentsLayout
-                            user={user}
-                            pillar={pillar}
-                            baseUrl={CAPI.config.discussionApiUrl}
-                            shortUrl={CAPI.config.shortUrlId}
-                            commentCount={commentCount}
-                            commentPage={commentPage}
-                            commentPageSize={commentPageSize}
-                            commentOrderBy={commentOrderBy}
-                            isClosedForComments={isClosedForComments}
-                            discussionD2Uid={CAPI.config.discussionD2Uid}
-                            discussionApiClientHeader={
-                                CAPI.config.discussionApiClientHeader
-                            }
-                            enableDiscussionSwitch={
-                                CAPI.config.enableDiscussionSwitch
-                            }
-                            expanded={false}
-                            commentToScrollTo={hashCommentId}
-                            onPermalinkClick={handlePermalink}
-                        />
-                    </Lazy>
-                )}
-            </Portal>
+            <Hydrate root="comments">
+                <CommentsLayout
+                    user={user}
+                    pillar={pillar}
+                    baseUrl={CAPI.config.discussionApiUrl}
+                    shortUrl={CAPI.config.shortUrlId}
+                    commentCount={commentCount}
+                    commentPage={commentPage}
+                    commentPageSize={commentPageSize}
+                    commentOrderBy={commentOrderBy}
+                    isClosedForComments={isClosedForComments}
+                    discussionD2Uid={CAPI.config.discussionD2Uid}
+                    discussionApiClientHeader={
+                        CAPI.config.discussionApiClientHeader
+                    }
+                    enableDiscussionSwitch={CAPI.config.enableDiscussionSwitch}
+                    expanded={!!openComments}
```

Comment:
`openComments` isn't very clear what it is from var name so the `!!` makes it even harder to reason with. Is it 'comments being open on the page' or 'comment box is open' or something else.

Reply:
Agreed 👍